mne/encoding/feature.py

```diff
@@ -0,0 +1,157 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+import numpy as np
+
+
+class DataDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    delays : array-like, shape (n_delays,)
+        The delays to create for the data. If sfreq is 1, it is assumed
+        that these are in samples. If sfreq != 1, these should be in
+        seconds.
+    sfreq : float | int
+        The sampling frequency to use when constructing delays. If 1, then
+        delays should contain sample indices, not time in seconds.
+    """
+    def __init__(self, delays=[0.], sfreq=1.):
+        self.delays = np.asarray(delays)
+        self.sfreq = float(sfreq)
+
+    def fit(self, X, X_names=None):
+        """Create a time-lagged representation of X.
```

Comment:
The name isn't necessary is it ? 

And this would intrinsically break the compatibility sklearn pipeline ?